spec/device_detector/device_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+describe DeviceDetector::Device do
+
+  subject(:device) { DeviceDetector::Device.new(user_agent) }
+
+  describe '#name' do
+
+    context 'when models are nested' do
+      let(:user_agent) { 'Mozilla/5.0 (iPhone; CPU iPhone OS 8_1_3 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Mobile/12B466 [FBDV/iPhone7,2]' }
+
+      it 'should find an Apple iPhone 6' do
```

Comment:
A bit stylistic thing, I like the expectations to be expressing themselves in a "confident" way, e.g `it 'finds an Apple iPhone6`, without the should.